src/tools/fep.py

```diff
@@ -418,6 +428,16 @@ def abfe_end_to_end(
             self._ligands_db.cols + self._proteins_db.cols + self._abfe_db.cols
         )
 
+        # only run on ligands that have not been run yet
+        # first check that there are no existing runs
+        df = api.get_dataframe(DB_ABFE)
+        df[df["Protein"] == self.protein._do_id]
+
+        df = df[(df[COL_LIGAND].isin(ligand_ids)) & (~pd.isna(df[COL_RESULT]))]
+
+        already_run_ligands = set(df[COL_LIGAND])
+        ligand_ids = set(ligand_ids) - already_run_ligands
+
```

Comment:
we prevent users from starting a run if that ligand/protein combo has already been run